### PR TITLE
Add golfer page - frontend

### PR DESCRIPTION
**Pull request by cristian-scarisoreanu**
### Changes

- Changed the `Layout` to render using a given id, which is used to create a link to the page of the logged-in golfer.
- Added a link to the home page by wrapping the _Home_ text in a `Link` component.
- Changed the `ScoreCard` by adding a link to every golfer's page.
- Added `/lib/useUserScores` to fetch a golfer's name and scores from the backend (using the specific API URL).
- Added the golfer page `pages/golfer/[id]` which displays the golfer's name, the number of scores he has, and a list of all of his scores ordered by date. On this page, the `ScorePostWidget` will only render for the logged-in user.

### Before

**Hompage (scores feed)**

image

**Golfer page**

image

### After

**Hompage (scores feed)**

image

**Page of the logged-in golfer**

image

**Page of a golfer with scores**

image

**Page of a golfer with no scores**

image

**Page of a nonexistent golfer/user**

image

### CODE REVIEW

File: components/ScoreCard.js [hunk 1]
```diff
@@ -14,7 +16,15 @@ const ScoreCard = ({ id, playedAt, totalScore, userId, userName }) => {
...
         <div>
-          {`${userName} posted a score of ${totalScore}`}
+          {useRouter().pathname !== '/golfers/[id]' && (
+            <Link href={`/golfers/${encodeURIComponent(userId)}`}>
+              <a className="hover:underline text-blue-700">{userName}</a>
+            </Link>
+          )}
+          {useRouter().pathname === '/golfers/[id]' && (
+            <span>{`${userName}`}</span>
+          )}
```
Comment (SergiuApostuTOL) [inline components/ScoreCard.js]: Avoid calling a hook multiple times if you don't need too.

Move the result of `useRouter()` to a constant/variable and use that one to get the path name.

File: lib/useUserScores [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import useSWR from 'swr'
```
Comment (SergiuApostuTOL) [inline lib/useUserScores]: This file has no extension 😄

File: components/Layout.js [hunk 3]
```diff
@@ -2,19 +2,24 @@ import Link from 'next/link'
 import { useState, useEffect } from 'react'
 import { getUsername } from '../lib/userAuth'
 
-const Layout = ({ children }) => {
+const Layout = ({ children, id }) => {
```
Comment (SergiuApostuTOL) [inline components/Layout.js]: Why do we need to receive the `id` from the parent? In both of the cases where we are using Layout, id is set to `getUserId()`. I think we can just call that method inside this component.

File: pages/golfers/[id].js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                  userId={score.user_id}
+                  userName={user.name}
+                />
+              ))}
+            </>
+          )}
+        </>
+      </Layout>
+    )
+  } else { return null }
```
Comment (SergiuApostuTOL) [inline pages/golfers/[id].js]: It's hard to see `return null` at the end of the file after many lines of JSX. Usually, this cases are handled earlier.

```js
if (!userId) {
  return null
}

return (
  <Layout>
    ...
  </Layout>
)
```

File: pages/index.js [hunk 5]
```diff
@@ -2,33 +2,36 @@ import Layout from '../components/Layout'
...
-                userId={score.user_id}
-                userName={score.user_name}
-              />
-            ))}
-          </>
-        )}
-      </>
-    </Layout>
-  )
+  if (scores || error) {
```
Comment (SergiuApostuTOL) [inline pages/index.js]: Why are we displaying this page only if there are scores or there is an error?

**Review by SergiuApostuTOL — CHANGES_REQUESTED** [review]

File: components/Layout.js (hunk 3 above)
Comment (cristian-scarisoreanu) [reply components/Layout.js]: I've added a custom hook to set the `userId` to the result of `getUserId()` method. In this way I will not get an error by calling `getUserId()` directly in the `Layout` component . (local storage is not defined)

image

**Review by SergiuApostuTOL — APPROVED** [review]

**Pull request merged**